### PR TITLE
Move annotation logic into Query Runner

### PR DESCRIPTION
**Pull request by arikfr**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description

This is a small refactor towards allowing controlling the query annotation in the query runner (a relevant use case is coming up in a follow up PR). 

In this pull request, we move the annotation logic into the query runner (with a default implementation in `BaseQueryRunner`).

### CODE REVIEW

File: redash/query_runner/pg.py [hunk 1]
```diff
@@ -38,10 +38,8 @@ def default(self, o):
```
Comment (arikfr) [inline redash/query_runner/pg.py]: Nothing changed here, just auto formatting.

**Comment by rauchy** [comment]
What's the benefit in implementing this as a mixin, as opposed to an actual method on `BaseQueryRunner`?

**Comment by arikfr** [comment]
> What's the benefit in implementing this as a mixin, as opposed to an actual method on `BaseQueryRunner`?

Because the one implemented in `BaseQueryRunner` is the implementation that does annotate, so we need the mixin to override it. 

Although now I wonder if we might get bitten by the order in which the methods get overriden.

**Comment by rauchy** [comment]
Yeah, I got that, but still -

(a) seems like _most_ query runners have no annotations (in this PR, you mix it in about 35 times).
(b) adding a mixin _to remove_ functionality feels counterintuitive.

Don't you think it'll be easier to understand if it were the other way around? (an `AnnotationMixin`)

**Comment by arikfr** [comment]
> seems like _most_ query runners have no annotations (in this PR, you mix it in about 35 times).

You should improve your grep condition :-) You probably included the `import` statements in this count. It's being used 27 times. We have 56 Query Runners, so that's 48% ;-) So we still have a majority that use annotations.

I still prefer the annotation to be the default, but here's an alternative approach: add a kill switch instead of mixin, so it becomes:

```python
class JSON(BaseHTTPQueryRunner):
    add_query_annotation = False
``` 

Might need a better name for the flag.

**Comment by rauchy** [comment]
> You should improve your grep condition :-) 

damnit you're right 😠 I still find a "mixins add behavior" approach simpler, but 🤷‍♂ 

Regarding the alternative - how about simply `annotate = True`.
Also, did you consider an `@annotated` decorator?

**Comment by arikfr** [comment]
I agree that it makes more sense for a mixin to add behavior. Although one could claim that this adds behavior too (of doing nothing). 🤪

> Regarding the alternative - how about simply `annotate = True`.

Feels a bit unclear, like "annotate what"? Ideally it would be `annotate_query`, but that's the method name :-(

**Comment by rauchy** [comment]
> Feels a bit unclear, like "annotate what"?
True dat.

> Also, did you consider an @annotated decorator?

**Comment by arikfr** [comment]
> > Also, did you consider an @annotated decorator?

Feel like an overkill for something that can be a simple boolean class property.

**Pull request merged**